src/Fixer/Whitespace/BlankLineBeforeControlStatementFixer.php

```diff
@@ -0,0 +1,269 @@
+    $foo->bar();
+    throw new \UnexpectedValueException("A cannot be null");
+}',
+                    [
+                        'statements' => ['throw'],
+                    ]
+                ),
+                new CodeSample(
+'<?php
+$a = 9000;
+try {
+    $foo->bar();
+} catch (\Exception $exception) {
+    $a = -1;
+}',
+                    [
+                        'statements' => ['try'],
+                    ]
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // should be run after NoUselessReturnFixer
+        return -19;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound(array_values($this->fixTokenMap));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $lineEnding = $this->whitespacesConfig->getLineEnding();
+        $tokenKinds = array_values($this->fixTokenMap);
+        $analyzer = new TokensAnalyzer($tokens);
+
+        for ($index = 0, $limit = $tokens->count(); $index < $limit; ++$index) {
+            $token = $tokens[$index];
+
+            if (!$token->isGivenKind($tokenKinds) || ($token->isGivenKind(T_WHILE) && $analyzer->isWhilePartOfDoWhile($index))) {
+                continue;
+            }
+
+            $prevNonWhitespaceToken = $tokens[$tokens->getPrevNonWhitespace($index)];
+
+            if (!$prevNonWhitespaceToken->equalsAny([';', '}'])) {
+                continue;
+            }
+
+            $prevIndex = $index - 1;
+            $prevToken = $tokens[$prevIndex];
+
+            if ($prevToken->isWhitespace()) {
+                $countParts = substr_count($prevToken->getContent(), "\n");
+
+                if (0 === $countParts) {
+                    $tokens[$prevIndex] = new Token([T_WHITESPACE, rtrim($prevToken->getContent(), " \t").$lineEnding.$lineEnding]);
+                } elseif (1 === $countParts) {
+                    $tokens[$prevIndex] = new Token([T_WHITESPACE, $lineEnding.$prevToken->getContent()]);
+                }
+            } else {
+                $tokens->insertAt($index, new Token([T_WHITESPACE, $lineEnding.$lineEnding]));
+
+                ++$index;
+                ++$limit;
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('statements', 'List of control statements to fix.'))
+                ->setAllowedTypes(['array'])
+                ->setAllowedValues([
+                    (new FixerOptionValidatorGenerator())->allowedValueIsSubsetOf(array_keys(self::$tokenMap)),
+                ])
+                ->setDefault([
+                    'break',
+                    'continue',
+                    'do',
```

Comment:
I still think `do`, `for`, `foreach`, `if`, `switch` and `while` should not be part of the default value. They are often preceded by variable initialization or something similar and adding a newline does not _always_ improve readability in such case IMO.

This statement could apply for other keywords, but less likely so I would keep them.